impl/maven-di/src/test/java/org/apache/maven/di/impl/InjectorImplTest.java

```diff
@@ -416,11 +416,20 @@ void testCircularPriorityDependency() {
         DIException exception = assertThrows(DIException.class, () -> {
             injector.getInstance(CircularPriorityTest.MyService.class);
         });
-        assertThat(exception).isInstanceOf(DIException.class).hasMessageContaining("HighPriorityServiceImpl");
-        assertThat(exception.getCause())
-                .isInstanceOf(DIException.class)
-                .hasMessageContaining("Cyclic dependency detected")
-                .hasMessageContaining("MyService");
+        assertInstanceOf(DIException.class, exception, "Expected exception to be DIException");
+        assertTrue(
+                exception.getMessage().contains("HighPriorityServiceImpl"),
+                "Expected exception message to contain 'HighPriorityServiceImpl' but was: " + exception.getMessage());
+
+        assertInstanceOf(DIException.class, exception.getCause(), "Expected cause to be DIException");
+        assertTrue(
+                exception.getCause().getMessage().contains("Cyclic dependency detected"),
+                "Expected cause message to contain 'Cyclic dependency detected' but was: "
+                        + exception.getCause().getMessage());
+        assertTrue(
+                exception.getCause().getMessage().contains("MyService"),
+                "Expected cause message to contain 'MyService' but was: "
+                        + exception.getCause().getMessage());
```

Comment:
for me assertJ was more readable and less code with descriptions - as result we have the same